genesiscommunitysuccess/custom-elements-lsp
# chore: setup analyzer plugin as typescript project FUI-1528

🤔  &nbsp; **What does this PR do?**

- Implements typescript in the analyzer plugin module
- Adds basic unit testing for analyzer plugin module
- This does *not* add in the actual analyzer plugin we need - that will be part of a subsequent PR!
- Bumps version of eslint prettier plugin which created some linting issues
- Fixes the linting issues in the lsp which arose from the above change (basically just adding loads of trailing commas)

📑  &nbsp; **How should this be tested?**

Update test steps to match your PR:

```
1. Checkout branch
2. `npm run bootstrap`
3. `npm run test`

To test the LSP is working correctly in VSCode
1. `cd packages/showcase/example`
2. `code .` (or manually open VSCode via the UI to the `example` directory on your filesystem).
3. Open a typescript file such as `root.ts`
4. In the Command Palette choose `TypeScript: select typescript version...` and then choose the workspace version
5. The LSP should be enabled. You will need to make a code change after the LSP is enabled before you see any Intellisense.
```

✅  &nbsp; **Checklist**
